### PR TITLE
Fix rebuild_exception_cache

### PR DESCRIPTION
**Pull request by Nyaran**
Related #6743 

Proposed changes in this pull request:
- Fix the rebuild. Now it really replaces the cached data
- Update `exceptionsSeasonCache`, just to ensure is updated after the rebuild.

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickChill/SickChill/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: sickchill/oldbeard/scene_exceptions.py [hunk 1]
```diff
@@ -324,14 +324,15 @@ def getSceneSeasons(indexer_id):
...
-        if result["season"] not in exceptionsCache[indexer_id]:
-            exceptionsCache[indexer_id][result["season"]] = []
+        if result["season"] not in exceptions_cache_list:
+            exceptions_cache_list[result["season"]] = []
+
+        exceptions_cache_list[result["season"]].append(result["show_name"])
 
-        exceptionsCache[indexer_id][result["season"]].append(result["show_name"])
+    exceptionsCache[indexer_id] = exceptions_cache_list
+    exceptionsSeasonCache[indexer_id] = exceptions_cache_list.keys()
```
Comment (miigotu) [inline sickchill/oldbeard/scene_exceptions.py:338]: Above this `exceptionsSeasonCache[indexer_id]` is a list of integers for each indexer_id, and the exceptionsCache keys are strings.

I think you need to:
`exceptionsSeasonCache[indexer_id] = list({int(x) for x in exceptions_cache_list.keys()})`

Comment (Nyaran) [reply sickchill/oldbeard/scene_exceptions.py:338]: That is not what I can see debuging

`exceptionsCache` keys are numbers, and that key is the indexer id.
`exceptionsCache[indexer_id]` also have keys, and that keys are numbers too, as the `season` column on `scene_exceptions` table

Comment (miigotu) [reply sickchill/oldbeard/scene_exceptions.py:338]: Yes, but are numbers but one is `"1"` (strings) and one is `1` (int)
I mean the object type not the value

Comment (miigotu) [reply sickchill/oldbeard/scene_exceptions.py:338]: Line 111-112
```python

if sql_results:
            exceptions_season_list = list({int(x["season"]) for x in sql_results})
```

Not sure which needs to change but the types need to be the same.

Comment (miigotu) [reply sickchill/oldbeard/scene_exceptions.py:338]: You might be right that the sql is returning the correct type for result["season"] now on python3 (it didn't used to).
I'll have to check myself later. 

Sent you an invite to the org btw

Comment (miigotu) [reply sickchill/oldbeard/scene_exceptions.py:338]: You were correct, sorry lol

**Comment by miigotu** [comment]
I refactored sceneSeason, pushing soon

**Pull request merged**